appstudio-controller/controllers/appstudio.redhat.com/snapshotenvironmentbinding_controller.go

```diff
@@ -551,10 +608,177 @@ func setLabel(deployment *apibackend.GitOpsDeployment, key, value string) error
 // SetupWithManager sets up the controller with the Manager.
 func (r *SnapshotEnvironmentBindingReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		// Uncomment the following line adding a pointer to an instance of the controlled resource as an argument
 		For(&appstudioshared.SnapshotEnvironmentBinding{}).
 		Owns(&apibackend.GitOpsDeployment{}).
-		Complete(r)
+		Watches(
+			&source.Kind{Type: &appstudioshared.Environment{}},
```

Comment:
Since we're now using the contents of Environments, DT, and DTC, in `generateExpectedGitOpsDeployment`, we need to watch them. 

HOPEFULLY this does not significantly increase the amount of work of our service, we can watch this after the code goes in.